include/flux/op/cartesian_base.hpp

```diff
@@ -222,7 +222,9 @@ struct cartesian_traits_base_impl {
                   && (sized_sequence<Bases> && ...))
     {
         return std::apply([](auto&... base) {
-            return (flux::size(base) * ...);
+            distance_t sz = 1;
```

Comment:
Though it should already be enforced by the public interface for `cartesian_product` and such, I wonder if there is value in static asserting `sizeof...(Bases) > 0` somewhere either in this function or in this struct given this default return value of 1?

Reply:
I've now changed the signature of the lambda to `[](auto& base0, auto&... bases)` which means we don't need to initialise `sz = 1`, and we get a check that the `self.base_` tuple has at least one element, the best of both worlds :)